### ISSUE TITLE
[Feature Request] Add option to disallow generic rule flags

### ISSUE THREAD

**Issue by pe8ter** [palantir/tslint#4361]
### Feature request

**Is your feature request related to a problem? Please describe.**
When using the generic versions of TSLint's rule flags, _all_ linting is off for the specified scope. It would be better to minimize the amount of code that is not linted by having the option to disallow the generic forms of rule flags.

For example, let's say a project has both `no-magic-numbers` and `no-bitwise` enabled. Now someone wants to temporarily disable `no-bitwise` like this:

```typescript
// tslint:disable-next-line
const x = foo & 4;
```

By using a generic rule flag, the code unnecessarily loses the `no-magic-numbers` flag and we don't know we're in violation of that rule since all linting is off. It's not clear that the intent of the rule flag was only to disable the `no-bitwise` rule.

**Describe the solution you'd like**
I propose a new configuration option to disallow generic rule flags. Something like this in _tslint.json_:

```json
{
    "linterOptions": {
        "strictRuleFlags": true
    }
}
```

With this option, the above example would require the specific rule to be called out. The intent of the flag would be documented more clearly and the efficacy of all linting would increase. This is ideal for large projects with many contributors.

**Describe alternatives you've considered**
Another option is to make a new rule that parses comments for the generic rule flags.

**Comment by pe8ter** [comment]
Apologies for not finding the other item.

There are two reasons I can think of to make this a TSLint configuration instead of a lint rule:

1. TSLint already parses comments for these rule flags so it's natural to extend that behavior. It seems like wasted effort to re-analyze comments for the same flags that TSLint already knows how find.
2. This feature is about how we use TSLint, not about how we write TypeScript code.

**Comment by pe8ter** [comment]
I tried creating this rule myself and I believe that it is not possible to have a rule that does this kind of linting. After TSLint builds a list of all rule violations it filters out any violations that have been disabled with the `tslint:disable` flag. Since this is the code that is meant to be linted, the rule flag essentially disables itself.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by crh225** [comment]
It looks like someone developed it on the Angular/cli project, but also in the same project, they are still able to disable, the disable:
![image](https://user-images.githubusercontent.com/19475195/67801811-f91c7000-fa57-11e9-9c59-84c879d903ac.png)

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
